### PR TITLE
DB set method

### LINKED ISSUE #131: DB set method
* Add a `set` method to `FirestoreDatabase` that add an item to the database with a specific ID instead of an auto-generated one
* Add a `set` method to `UserTable` to create users with their uid (coming from Firebase authentication)

### PR DESCRIPTION
**Pull request by Prolas**
Created setItem method in FirestoreDatabase.kt
If an id is given the item is set at this id otherwise a new id is generated (same as add)
Modified UserTableUnitTest to test it

close #131

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt [hunk 1]
```diff
@@ -62,6 +62,40 @@ class FirestoreDatabase(private val useEmulator: Boolean = false) {
...
+   *
+   * @param path A filesystem-like path that specify the location of the table
+   * @param item The item to set to the database, the types of the attributes have to be Firestore
+   * @param id the id of the item (if null a new id is generated)
+   * @param onCompletion Callback called on completion of the operation
+   * @param onError Callback called when an error occurs
+   */
+  fun <T : Any> setItem(
+      path: String,
+      id: String?,
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt:78]: In the method `setItem`, you expect to receive an `id`, so you should mark the `id` argument as non-nullable (that way you don't find yourself calling `additem`)

File: app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt [hunk 2]
```diff
@@ -62,6 +62,40 @@ class FirestoreDatabase(private val useEmulator: Boolean = false) {
...
+      item: T,
+      onCompletion: (id: String) -> Unit,
+      onError: (Exception) -> Unit
+  ) {
+    if (id != null) {
+      db.collection(path)
+          .add(item)
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt:85]: Pretty sure it should not be `add` (in your whole method you never use the `id` argument so there definitely something fishy...)

File: app/src/main/java/ch/epfl/skysync/database/tables/UserTable.kt [hunk 3]
```diff
@@ -97,6 +97,25 @@ class UserTable(db: FirestoreDatabase) : Table<User, UserSchema>(db, UserSchema:
...
+   *
+   * Set item at id (if null a new id is generated) and override any previously set id.
+   *
+   * @param item The user to add to the database
+   * @param id the id of the item (if null a new id is generated)
+   * @param onCompletion Callback called on completion of the operation
+   * @param onError Callback called when an error occurs
+   */
+  fun set(
+      id: String?,
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/database/tables/UserTable.kt:111]: Same comment as above, `id` should be non-nullable

**Review by Plouc314 — CHANGES_REQUESTED** [review]
Small oversight mistakes but you got the main idea of the database

File: app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt (hunk 2 above)
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt:85]: Indeed this should be .document(id).set(item)

File: app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt (hunk 1 above)
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt:78]: OK, thanks. I was wondering if the id could be null, and if so, I didn't see the difference with add.

**Review by Plouc314 — APPROVED** [review]
LGTM

**Pull request merged**